src/components/Modal/ModalReviewSwap.tsx

```diff
@@ -39,45 +40,75 @@ const ModalReviewSwap = () => {
   const { onCloseModal, setModalType, payload } = useModalStore(
     (state) => state
   )
-  const { getAccountBalance } = useAccountBalance()
-  const { getSwapEstimateBot, isFetching } = useSwapEstimateBot()
-
+  const [isFetching, setIsFetching] = useState(false)
+  const { getSwapEstimateBot } = useSwapEstimateBot()
   const [convertPayload, setConvertPayload] = useState<ModalReviewSwapPayload>(
     payload as ModalReviewSwapPayload
   )
+  const {
+    priceToUsd: priceToUsdTokenIn,
+    calculateTokenToUsd: calculateTokenToUsdTokenIn,
+  } = useCalculateTokenToUsd()
+  const {
+    priceToUsd: priceToUsdTokenOut,
+    calculateTokenToUsd: calculateTokenToUsdTokenOut,
+  } = useCalculateTokenToUsd()
+  const { getAccountBalance } = useAccountBalance()
   const [isWNearConjunctionRequired, setIsWNearConjunctionRequired] =
     useState(false)
 
   const recalculateEstimation = async () => {
-    const pair = [
-      convertPayload.selectedTokenIn.address as string,
-      convertPayload.selectedTokenOut.address as string,
-    ]
-    // Not needed recalculation if ratio is 1:1
-    if (pair.includes("native") && pair.includes("wrap.near")) return
-
-    handleCheckNativeBalance()
+    try {
```

Comment:
You don't need this `try {} catch {}` cause we use React Query which cover this everything and you can get this from response, at `useSwapEstimateBot`, and for most queries, it's usually sufficient to check for the isLoading state, then the isError state, then finally, assume that the data is available and render the successful state